10_code/threshold.py

```diff
@@ -143,72 +148,92 @@
 
 #add fips to population
-pop_tx_fips = pd.merge(fips_tx, pop_tx_merged, on = "County", how = "left", indicator=True)
+pop_tx_fips = pd.merge(fips_tx, pop_tx_merged, on = ["state","County"], how = "left", indicator=True)
 pop_tx_fips["_merge"].value_counts() #✅ both 
 pop_tx_final = pop_tx_fips[["state", "fips", "County", "year", "population"]].copy()
 
 #reintroduce overdose data
 
-overdose_tx = deaths_tx[(~(deaths_tx["Year"] == 2015))].copy()
-overdose_tx.rename(columns={"County Code": "fips","State":"state"}, inplace=True)
-overdose_tx["fips"] = overdose_tx["fips"].astype(np.int64)
-overdose_tx["year"] = overdose_tx["Year"].astype(np.int64)
-overdose_tx.sort_values(by = ["state", "fips"], inplace=True)
+overdose_deaths_tx = deaths_tx[(~(deaths_tx["Year"] == 2015))].copy()
+overdose_deaths_tx.rename(columns={"County Code": "fips","State":"state"}, inplace=True)
+overdose_deaths_tx["fips"] = overdose_deaths_tx["fips"].astype(np.int64)
+overdose_deaths_tx["year"] = overdose_deaths_tx["Year"].astype(np.int64)
+overdose_deaths_tx.sort_values(by = ["state", "fips"], inplace=True)
 
-#overdose 2003 to 2005
-overdose_tx_03 = overdose_tx[(overdose_tx["year"] <= 2005)]
+# #overdose 2003 to 2005
+# overdose_deaths_tx_03 = overdose_deaths_tx[(overdose_deaths_tx["year"] <= 2007)]
 
-#overdose 2006 to 2011
-overdose_tx_06 = overdose_tx[(overdose_tx["year"]>= 2006)]
+# #overdose 2006 to 2014
+# overdose_deaths_tx_08 = overdose_deaths_tx[(overdose_deaths_tx["year"]>= 2008)]
 
 
 
 #check for counties with full overdose data
-fips_tx_group = overdose_tx_06["fips"].unique()
-
-
-counties_full_overdose_data = []
-for county in fips_tx_group:
-    tmp = overdose_tx_06[(overdose_tx_06["fips"] == county)].copy()
-    if len(tmp["year"]) == 9:
-        tmp1 = pop_tx_fips[(pop_tx_fips["fips"] == county)]
-        #assert len(tmp1["year"]) == 12
-        tmp_merge = pd.merge(tmp, tmp1, on = "year", how = "left")
-        counties_full_overdose_data.append(tmp_merge)
-        pass
-
+# fips_tx_group = overdose_deaths_tx_08["fips"].unique()
+
+
+# counties_full_overdose_data = []
+# for county in fips_tx_group:
+#     tmp = overdose_deaths_tx_08[(overdose_deaths_tx_08["fips"] == county)].copy()
+#     if len(tmp["year"]) == 9:
+#         tmp1 = pop_tx_fips[(pop_tx_fips["fips"] == county)]
+#         #assert len(tmp1["year"]) == 12
+#         tmp_merge = pd.merge(tmp, tmp1, on = "year", how = "left")
+#         counties_full_overdose_data.append(tmp_merge)
+#         pass
+
+#final data to use
+tx_merge_final= pd.merge(overdose_deaths_tx,pop_tx_final, on = ["fips","year"], how="left", indicator=True)
+tx_merge_final["_merge"].value_counts()
+tx_final_data= tx_merge_final[["state_x", "fips", "year", "County" , "population" , "deaths"]].copy()
+tx_final_data.rename(columns={"state_x" : "state"}, inplace=True)
+#policy change year for texas 2007
+#pre data
+tx_prepolicy= tx_final_data[(tx_final_data["year"] <= 2006)]
```

Comment:
While testing this out with Emma, we realized there is no need to split pre and post. Just write one csv for Texas.